packages/core/test/coordinator-test.js

```diff
@@ -11,4 +14,26 @@ describe('coordinator', () => {
 
     assert.strictEqual(mc2, coordinator());
   });
+
+  it('query results returned in correct order', async () => {
+
+    const connector = {
+      uid : 0,
+
+      async query(query) {
+        this.uid++;
+        let id = this.uid;
+        //Earlier queries are resolved later so that they are purposely returned out of order
+        let result = new Promise((resolve, reject) => setTimeout(() => resolve(id), 200*(10-id)));
```

Comment:
Let's not use timeout in tests because it slows down the tests. Instead figure out a way to resolve the promises out of order.